File: AdHoc/2590.py
n = int(raw_input())

i = 0
while i < n: 
	entrada = input()

	if entrada % 4 == 0:
		print "1"
	elif entrada % 4 == 1:
		print "7"
	elif entrada % 4 == 2:
		print "9"
	elif entrada % 4 == 3:
		print "3"

	i = i + 1

File: AdHoc/2590.c
#include <stdio.h>

int main(void)
{
	int entrada, n, i;
	scanf("%d", &n);

	for(i = 0; i < n; i++)
	{
		scanf("%d", &entrada);

		if (entrada % 4 == 0)
			printf("1\n");
		else if (entrada % 4 == 1)
			printf("7\n");
		else if (entrada % 4 == 2)
			printf("9\n");
		else if (entrada % 4 == 3)
			printf("3\n");
	}

}